src/vanilla/utils/atomWithStorage.ts

```diff
@@ -99,8 +99,18 @@ export function createJSONStorage<Value>(
   getStringStorage: () =>
     | AsyncStringStorage
     | SyncStringStorage
-    | undefined = () =>
-    typeof window !== 'undefined' ? window.localStorage : undefined,
+    | undefined = () => {
+    try {
+      return window.localStorage
+    } catch (e) {
+      // Window could be undefined or local storage could be disabled
+      if (e instanceof TypeError || e instanceof DOMException) {
+        return undefined
+      }
+
+      throw e
+    }
```

Comment:
completely different idea:
```suggestion
    try {
      return window.localStorage
    } catch (e) {
      if (import.meta.env?.MODE !== 'production') {
        if (typeof window !== 'undefined') {
          console.warn(e);
        }
      }
      return undefined
    }
```